cassandra-mesos-executor/src/test/java/io/mesosphere/mesos/frameworks/cassandra/executor/BackupManagerTest.java

```diff
@@ -0,0 +1,191 @@
+        File tableDir = new File(backupDir, keyspace + "/" + table);
+        assertTrue(tableDir.mkdirs());
+        Files.createFile(new File(tableDir, "data.db").toPath());
+    }
+
+    private void delete(File file) {
+        if (file.isDirectory()) {
+            File[] children = file.listFiles();
+
+            if (children != null) {
+                for (File child: children) {
+                    delete(child);
+                }
+            }
+        }
+
+        assertTrue(file.delete());
+    }
+
+    class TestJmxConnect implements JmxConnect {
+        private String dataDir;
+        private List<String> invocations = new ArrayList<String>();
+
+        TestJmxConnect(String dataDir) {
+            this.dataDir = dataDir;
+        }
+
+        public List<String> getInvocations() { return Collections.unmodifiableList(invocations); }
+
+        @NotNull
+        @Override
+        public RuntimeMXBean getRuntimeProxy() {
+            throw new UnsupportedOperationException();
+        }
+
+        @NotNull
+        @Override
+        public StorageServiceMBean getStorageServiceProxy() {
+            return newStorageService();
+        }
+
+        @NotNull
+        @Override
+        public EndpointSnitchInfoMBean getEndpointSnitchInfoProxy() {
+            throw new UnsupportedOperationException();
+        }
+
+        @NotNull
+        @Override
+        public ColumnFamilyStoreMBean getColumnFamilyStoreProxy(@NotNull String keyspace, @NotNull String table) {
+            return newColumnFamilyStore();
+        }
+
+        @NotNull
+        @Override
+        public List<String> getColumnFamilyNames(@NotNull String keyspace) {
+            return Arrays.asList(TABLE);
+        }
+
+        @Override
+        public void close() throws IOException {}
+
+        private StorageServiceMBean newStorageService() {
+            return (StorageServiceMBean) Proxy.newProxyInstance(getClass().getClassLoader(), new Class[]{StorageServiceMBean.class},
+                    new InvocationHandler() {
+                        @Override
+                        public Object invoke(Object proxy, Method method, Object[] args) throws Throwable {
+                            String name = method.getName();
+                            switch (name) {
+                                case "takeSnapshot":
+                                    invocations.add("takeSnapshot");
+                                    return null;
+                                case "clearSnapshot":
+                                    invocations.add("clearSnapshot");
+                                    return null;
+                                case "truncate":
+                                    invocations.add("truncate");
+                                    return null;
+                                case "getAllDataFileLocations":
+                                    return new String[]{dataDir};
```

Comment:
Let's keep this as a single string for now until we actually add the support for multiple locations. It is confusing hard coding index 0 elsewhere.